File: scripts/extract.py
import zipfile
import os

zip_path = '/vercel/share/v0-project/package-lock.zip'
extract_dir = '/vercel/share/v0-project'

try:
    print(f'[v0] Extracting {zip_path}')
    print(f'[v0] To directory: {extract_dir}')
    
    if not os.path.exists(zip_path):
        raise FileNotFoundError(f'Zip file not found at {zip_path}')
    
    with zipfile.ZipFile(zip_path, 'r') as zip_ref:
        zip_ref.extractall(extract_dir)
    
    print('[v0] Extraction successful!')
    
except Exception as e:
    print(f'[v0] Error: {str(e)}')
    exit(1)

File: scripts/extract.js
import { createWriteStream, createReadStream } from 'fs';
import { extractSync } from 'bun';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const projectRoot = path.resolve(__dirname, '..');
const zipPath = path.join(projectRoot, 'package-lock.zip');

try {
  console.log('[v0] Starting extraction from:', zipPath);
  console.log('[v0] Extracting to:', projectRoot);
  
  // Use bun's extractSync to extract the zip
  extractSync({
    file: zipPath,
    dir: projectRoot,
  });
  
  console.log('[v0] Extraction successful!');
} catch (error) {
  console.error('[v0] Error extracting file:', error.message);
  process.exit(1);
}
